### PR TITLE
Add Claim and Swap Example

### PR DESCRIPTION
**Pull request by yvesfracari**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,7 +66,8 @@ forge test -vvv --fork-url https://eth.llamarpc.com
 
 Two examples are included for reference:
 1. [`./examples/mintDaiAndSwap.ts`](./examples/mintDaiAndSwap.ts) - In this example, the user approves the proxy contract to take actions on its behalf on the maker protocol and uses prehooks to just-in-time(JIT) borrow DAI right before the DAI gets swapped to COW.
-1. [`./examples/swapAndBridge.ts`](./examples/swapAndBridge.ts) - In this example, the user uses the proxy address as the receiver for the swapped tokens and in the posthook it bridges the exact amount of swap output([weiroll](https://github.com/weiroll/weiroll) is used for this) to gnosis chain with user's address as the recipient.
+2. [`./examples/swapAndBridge.ts`](./examples/swapAndBridge.ts) - In this example, the user uses the proxy address as the receiver for the swapped tokens and in the posthook it bridges the exact amount of swap output([weiroll](https://github.com/weiroll/weiroll) is used for this) to gnosis chain with user's address as the recipient.
+3. [`./examples/swapAndBridge.ts`](.examples/claimAndSwap) - In this example, the user claim a WETH from Llama Pay from a vesting contract using a prehook right before swap it to CoW.
```
Comment (meetmangukiya) [inline README.md:70]:
```suggestion
3. [`./examples/swapAndBridge.ts`](.examples/claimAndSwap) - In this example, the user claims WETH from a Llama Pay vesting contract using a prehook right before swapping it to COW.
```
does this read better?

**Review by meetmangukiya — APPROVED** [review]
didn't run it locally but looks good code wise

File: README.md [hunk 2]
```diff
@@ -66,7 +66,8 @@ forge test -vvv --fork-url https://eth.llamarpc.com
 
 Two examples are included for reference:
 1. [`./examples/mintDaiAndSwap.ts`](./examples/mintDaiAndSwap.ts) - In this example, the user approves the proxy contract to take actions on its behalf on the maker protocol and uses prehooks to just-in-time(JIT) borrow DAI right before the DAI gets swapped to COW.
-1. [`./examples/swapAndBridge.ts`](./examples/swapAndBridge.ts) - In this example, the user uses the proxy address as the receiver for the swapped tokens and in the posthook it bridges the exact amount of swap output([weiroll](https://github.com/weiroll/weiroll) is used for this) to gnosis chain with user's address as the recipient.
+2. [`./examples/swapAndBridge.ts`](./examples/swapAndBridge.ts) - In this example, the user uses the proxy address as the receiver for the swapped tokens and in the posthook it bridges the exact amount of swap output([weiroll](https://github.com/weiroll/weiroll) is used for this) to gnosis chain with user's address as the recipient.
+3. [`./examples/swapAndBridge.ts`](.examples/claimAndSwap) - In this example, the user claims WETH from a Llama Pay vesting contract using a prehook right before swapping it to COW.
```
Comment (ribeirojose) [inline README.md:70]:
```suggestion
3. [`./examples/claimAndSwap.ts`](.examples/claimAndSwap) - In this example, the user claims WETH from a Llama Pay vesting contract using a prehook right before swapping it to COW.
```

File: examples/claimAndSwap.ts [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    feeAmount: 0n,
+    kind: OrderKind.SELL,
+    partiallyFillable: true,
+    sellTokenBalance: OrderBalance.ERC20,
+    buyTokenBalance: OrderBalance.ERC20,
+  };
+
+  // pre-hooks
+  const calls: ICall[] = [
+    // approve the bridge to spend the swapped usdc
```
Comment (mfw78) [inline examples/claimAndSwap.ts:105]: comment requires updating

File: examples/claimAndSwap.ts [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    userAddr,
+    encodedSignature
+  );
+
+  const hooks = {
+    pre: [
+      {
+        target: factory,
+        callData: hooksCalldata,
+        gasLimit: "9999999999999999999999",
```
Comment (mfw78) [inline examples/claimAndSwap.ts:148]: Being overzealous like this with gas limits may result in the API refusing the order (this gas limit is beyond block limits - but also likely beyond the ETH required when simulating settlement).

**Comment by mfw78** [comment]
Out of curiosity for this, do you have a reference transaction hash or similar for where this has been tested?

**Review by mfw78 — APPROVED** [review]
Looks good. One thing to note in the `README.md` or the example's header is that this example actually _isn't_ an example of an authenticated hook (i.e. the `claim(address)` function on the vesting contract is permissionless).

So this example could be implemented using normal pre-hooks without the additional gas overhead of going through cow-shed.

**Pull request merged**